pinot-server/src/main/java/org/apache/pinot/server/starter/helix/SegmentOnlineOfflineStateModelFactory.java

```diff
@@ -85,7 +85,16 @@ public void onBecomeConsumingFromOffline(Message message, NotificationContext co
     public void onBecomeOnlineFromConsuming(Message message, NotificationContext context)
         throws Exception {
       _logger.info("SegmentOnlineOfflineStateModel.onBecomeOnlineFromConsuming() : {}", message);
-      _instanceDataManager.addOnlineSegment(message.getResourceName(), message.getPartitionName());
+
+      try {
```

Comment:
Let's add it to all state transitions. We can consider making it a separate PR just for this